code/modules/modular_computers/networking/device_types/_network_device.dm

```diff
@@ -196,6 +221,156 @@
+	if(istype(holder, /obj/machinery))
+		return holder
+
+// Return the public methods and variables available for commands.
+/datum/extension/network_device/proc/get_public_methods()
+	var/obj/machinery/M = get_command_target()
+	if(istype(M))
+		return M.public_methods
+
+/datum/extension/network_device/proc/get_public_variables()
+	var/obj/machinery/M = get_command_target()
+	if(istype(M))
+		return M.public_variables
+
+/datum/extension/network_device/proc/set_command_reference(list/selected_commands, alias, reference)
+	if(!alias || !reference)
+		return
+	LAZYSET(selected_commands, alias, reference)
+
+/datum/extension/network_device/proc/add_command(list/selected_commands, alias, list/valid_commands)
+	if(!alias)
+		alias = copytext(md5(num2text(rand(0, 25))), 1, 11)
+	LAZYSET(selected_commands, alias, valid_commands[pick(valid_commands)])  // Random key and command
+	return alias
+
+/datum/extension/network_device/proc/change_command_alias(old_alias, new_alias)
+	new_alias = sanitize(new_alias)
+	new_alias = replacetext(new_alias, " ", "") // Strip spaces from the key.
+	
+	// Check if a command with the new alias already exists
+	if(LAZYACCESS(command_and_call, new_alias) || LAZYACCESS(command_and_write, new_alias))
+		return FALSE
+	
+	if(LAZYACCESS(command_and_call, old_alias))
+		LAZYSET(command_and_call, new_alias, LAZYACCESS(command_and_call, old_alias))
+		LAZYREMOVE(command_and_call, old_alias)
+		return TRUE
+
+	if(LAZYACCESS(command_and_write, old_alias))
+		LAZYSET(command_and_write, new_alias, LAZYACCESS(command_and_write, old_alias))
+		LAZYREMOVE(command_and_write, old_alias)
+		return TRUE
+
+/datum/extension/network_device/proc/remove_command(alias)
+	if(LAZYACCESS(command_and_call, alias))
+		LAZYREMOVE(command_and_call, alias)
+		return TRUE
+	if(LAZYACCESS(command_and_write, alias))
+		LAZYREMOVE(command_and_write, alias)
+		return TRUE
+	return FALSE
+
+// Any modification of public vars and calling methods from network commands. Return text feedback on success/unsuccess of command.
+/datum/extension/network_device/proc/on_command(command, command_args, mob/user)
+	var/datum/command_target = get_command_target()
+	if(!has_commands)
+		return "Device cannot receive commands."
+	if(!command_target)
+		return "No valid target found for command '[command]'"
+	if(!has_access(user))
+		return "Access denied."
+	if(command_and_call[command])
+		var/decl/public_access/public_method/method = command_and_call[command]
+		method.perform(command_target, command_args)
+		return "Successfully called method '[command]' on [network_tag]."
+	if(command_and_write[command])
+		var/decl/public_access/public_variable/variable = command_and_write[command]
+		if(command_args) // Write to a var.
+			command_args = sanitize_command_args(command_args, variable.var_type)
+			if(isnull(command_args)) // Command could not be converted into a form appropriate for the variable.
+				return "Improper argument type; correct type is [variable.var_type]"
```

Comment:
It's possible a null literal is a valid argument in some cases, though am not sure. Just be careful if you move this off to vars, as e.g. radio stuff may be passing nulls into num or text stuff.

Reply:
Looking through the writable vars, it doesn't look like any take in null as a possible value. The actual public variable var-type checking I added should still permit it, though, if the var-type is IC_FORMAT_ANYTHING